mcu/atmega-hal/src/usart.rs

```diff
@@ -17,17 +17,62 @@ avr_hal_generic::impl_usart_traditional! {
     tx: port::PD1,
 }
 
-#[cfg(feature = "atmega32u4")]
+#[cfg(any(feature = "atmega32u4", feature = "atmega2560"))]
 pub type Usart1<CLOCK, IMODE> = Usart<
     crate::RawPeripheral<crate::pac::USART1>,
     port::Pin<port::mode::Input<IMODE>, port::PD2>,
     port::Pin<port::mode::Output, port::PD3>,
     CLOCK,
 >;
-#[cfg(feature = "atmega32u4")]
+#[cfg(any(feature = "atmega32u4", feature = "atmega2560"))]
 avr_hal_generic::impl_usart_traditional! {
     peripheral: crate::RawPeripheral<crate::pac::USART1>,
     register_suffix: 1,
     rx: port::PD2,
     tx: port::PD3,
 }
+
+#[cfg(feature = "atmega2560")]
+pub type Usart0<CLOCK, IMODE> = Usart<
+    crate::RawPeripheral<crate::pac::USART0>,
+    port::Pin<port::mode::Input<IMODE>, port::PE0>,
+    port::Pin<port::mode::Output, port::PE1>,
+    CLOCK,
+>;
+#[cfg(feature = "atmega2560")]
+avr_hal_generic::impl_usart_traditional! {
+    peripheral: crate::RawPeripheral<crate::pac::USART0>,
+    register_suffix: 0,
+    rx: port::PE0,
+    tx: port::PE1,
+}
+
+#[cfg(feature = "atmega2560")]
+pub type Usart2<CLOCK, IMODE> = Usart<
+    crate::RawPeripheral<crate::pac::USART2>,
+    port::Pin<port::mode::Input<IMODE>, port::PH0>,
+    port::Pin<port::mode::Output, port::PH1>,
+    CLOCK,
+>;
+#[cfg(feature = "atmega2560")]
+avr_hal_generic::impl_usart_traditional! {
+    peripheral: crate::RawPeripheral<crate::pac::USART2>,
+    register_suffix: 2,
+    rx: port::PH0,
+    tx: port::PH1,
+}
+
+#[cfg(feature = "atmega2560")]
+pub type Usart3<CLOCK, IMODE> = Usart<
+    crate::RawPeripheral<crate::pac::USART3>,
+    port::Pin<port::mode::Input<IMODE>, port::PJ0>,
+    port::Pin<port::mode::Output, port::PJ1>,
+    CLOCK,
+>;
+#[cfg(feature = "atmega2560")]
+avr_hal_generic::impl_usart_traditional! {
+    peripheral: crate::RawPeripheral<crate::pac::USART3>,
+    register_suffix: 2,
+    rx: port::PJ0,
+    tx: port::PJ1,
+}
```

Comment:
Getting lots of compiler errors from this:

```text
error[E0609]: no field `ubrr2` on type `USART3`
  --> mcu/atmega-hal/src/usart.rs:73:1
   |
73 | / avr_hal_generic::impl_usart_traditional! {
74 | |     peripheral: crate::RawPeripheral<crate::pac::USART3>,
75 | |     register_suffix: 2,
76 | |     rx: port::PJ0,
77 | |     tx: port::PJ1,
78 | | }
   | |_^ unknown field
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

You can fix it with the following change:

```suggestion
#[cfg(feature = "atmega2560")]
avr_hal_generic::impl_usart_traditional! {
    peripheral: crate::RawPeripheral<crate::pac::USART3>,
    register_suffix: 3,
    rx: port::PJ0,
    tx: port::PJ1,
}
```